From dd90faeddf9074e814dbc44d201ee131af198741 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Wed, 22 Mar 2017 01:26:08 +0100
Subject: [PATCH] st: Add missing %NULL check

st_theme_node_get_border_image() may return %NULL, leading to a
segfault in st_border_image_get_file() when glib is compiled with
G_DISABLE_CHECKS.
---
 src/st/st-widget.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/st/st-widget.c b/src/st/st-widget.c
index 002249d25..db984ac9b 100644
--- a/src/st/st-widget.c
+++ b/src/st/st-widget.c
@@ -290,6 +290,7 @@ st_widget_texture_cache_changed (StTextureCache *cache,
   StWidget *actor = ST_WIDGET (user_data);
   StWidgetPrivate *priv = st_widget_get_instance_private (actor);
   StThemeNode *node = priv->theme_node;
+  StBorderImage *border_image;
   gboolean changed = FALSE;
   GFile *theme_file;
 
@@ -303,7 +304,8 @@ st_widget_texture_cache_changed (StTextureCache *cache,
       changed = TRUE;
     }
 
-  theme_file = st_border_image_get_file (st_theme_node_get_border_image (node));
+  border_image = st_theme_node_get_border_image (node);
+  theme_file = border_image ? st_border_image_get_file (border_image) : NULL;
   if ((theme_file != NULL) && g_file_equal (theme_file, file))
     {
       st_theme_node_invalidate_border_image (node);
-- 
2.12.0

